django_mongodb/compiler.py

```diff
@@ -412,6 +422,79 @@ def collection_name(self):
+                        (
+                            *self.query.extra_select,
+                            *self.query.values_select,
+                            *self.query.annotation_select,
+                        )
+                    )
+                compiler_.pre_sql_setup()
+                # Standardize columns as main query required.
+                columns = compiler_.get_columns()
+                if self.query.values_select:
+                    _, exprs = zip(*columns, strict=True)
+                    columns = tuple(zip(main_query_fields, exprs, strict=True))
+                parts.append((compiler_.build_query(columns), compiler_.collection_name))
+
+            except EmptyResultSet:
+                # Omit the empty queryset with UNION and with DIFFERENCE if the
+                # first queryset is nonempty.
+                if self.query.combinator == "union":
+                    continue
+                raise
+        # Raise EmptyResultSet if all the combinator queries are empty.
+        if not parts:
+            raise EmptyResultSet
+        combinator_pipeline = parts.pop(0)[0].get_pipeline()
+        if self.query.combinator == "union":
+            for part, collection in parts:
+                combinator_pipeline.append(
+                    {"$unionWith": {"coll": collection, "pipeline": part.get_pipeline()}}
+                )
+                if not self.query.combinator_all:
+                    ids = {}
+                    for alias, expr in main_query_columns:
+                        collection = expr.alias if isinstance(expr, Col) else None
+                        if collection and collection != self.collection_name:
+                            ids[
+                                f"{expr.alias}{self.GROUP_SEPARATOR}{expr.target.column}"
+                            ] = expr.as_mql(self, self.connection)
+                        else:
+                            ids[alias] = f"${alias}"
+                    combinator_pipeline.append({"$group": {"_id": ids}})
+                    projected_fields = defaultdict(dict)
+                    for key in ids:
+                        value = f"$_id.{key}"
+                        if self.GROUP_SEPARATOR in key:
+                            table, field = key.split(self.GROUP_SEPARATOR)
+                            projected_fields[table][field] = value
+                        else:
+                            projected_fields[key] = value
+                    # Convert defaultdict to dict so it doesn't appear as
+                    # "defaultdict(<CLASS 'dict'>, ..." in query logging.
+                    combinator_pipeline.append({"$addFields": dict(projected_fields)})
+                    if "_id" not in projected_fields:
+                        combinator_pipeline.append({"$unset": "_id"})
+        else:
+            raise NotSupportedError(f"Combinator {self.query.combinator} isn't supported.")
```

Comment:
Isn't this already checked by supports_select_{self.query.combinator}?

Reply:
Yes, when I wrote this code I had in mind the other combinators. But yes, it is being checked twice.